File: 100 problem/1reverse.js
//  How do you reverse a string in JavaScript?

//  Method 1
function reverseString(str) {
    return str.split("").reverse().join("");
}


// Method 2
function reverseString(str) {
    let i = str.length  - 1;
    new_str = ""

    while( i >= 0) {
        new_str += str[i];
        i--;
    }
    return new_str;
}

// using recursion | Method 3
function reverseString(str) {
    if(str === "") {
        return "";
    } else {
        return reverseString(str.substr(1)) + str.charAt(0);
    }
}


let str = "HELLO WORLD"
console.log(reverseString(str))

File: 100 problem/1reverse.py
# How do you reverse a string in Python?

# method 1
def reverse_str(str):
    new_str = ""
    for i in str:
        new_str = i  + new_str
        print(new_str + " ")
    return new_str

# Method 2
def reverse_str(str):
    return str[::-1]

str = input()
rev_str = reverse_str(str)
print(rev_str)